crates/fsrt/src/main.rs

```diff
@@ -99,7 +103,69 @@ impl fmt::Display for Error {
     }
 }
 
-impl std::error::Error for Error {}
+fn check_graphql_and_perms(val: &Value) -> Vec<&str> {
+    let mut operations = vec![];
+    match val {
+        Value::Const(Const::Literal(s)) => operations.extend_from_slice(&parse_graphql(s)),
+        Value::Phi(vals) => vals.iter().for_each(|val| match val {
+            Const::Literal(s) => operations.extend_from_slice(&parse_graphql(s)),
+        }),
+        _ => {}
+    }
+    // TODO : Build out permission resolver here
+
+    let permissions_resolver: HashMap<(&str, &str), &str> =
+        [(("compass", "searchTeams"), "read:component:compass")]
+            .iter()
+            .cloned()
+            .collect();
+
+    operations
+        .iter()
+        .filter(|f| permissions_resolver.contains_key(f))
+        .map(|f| permissions_resolver.get(f).unwrap().to_owned())
```

Comment:
```suggestion
        .filter_map(|f| permissions_resolver.get(f))
```

Reply:
I spent a little while looking at this and I initially tried to do the func mut, but the permissions_resolver.get(f) returns a `&&str` when we need a `&str`. AFAIK, there is no way to convert within the filter_map right?

Reply:
You can just add a `Option::copied` call after the `get`.